app/scripts/lib/SnapsNameProvider.ts

```diff
@@ -134,9 +134,12 @@ export class SnapsNameProvider implements NameProvider {
         },
       )) as AddressLookupResult;
 
-      const domain = result?.resolvedDomain;
+      const domains = result?.resolvedDomains;
 
-      proposedNames = domain ? [domain] : [];
+      // TODO: Determine if this is what we want.
+      proposedNames = domains
+        ? domains.map((domain) => domain.resolvedDomain)
+        : [];
```

Comment:
```suggestion
      proposedNames = domains
        ? [...new Set(domains.map((domain) => domain.resolvedDomain))]
        : [];
```

The snap can potentially (erroneously) return dupes, we should guard against that.